freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,437 @@
+            problem_progress=self.get_problem_progress(),
+            used_attempts_feedback=self.get_used_attempts_feedback(),
+            submit_class=self.get_submit_class(),
+            indicator_visibility_class=self.get_indicator_visiblity_class(),
+            word_count_message=self.get_word_count_message(),
+        )
+        fragment = self.build_fragment(
+            html_source=view_html,
+            paths_css=[
+                'view.less.min.css',
+            ],
+            paths_js=[
+                'view.js.min.js',
+            ],
+            fragment_js='FreetextresponseView',
+        )
+        return fragment
+
+    def studio_view(self, context=None):  # pylint: disable=unused-argument
+        '''
+        Build the fragment for the edit/studio view
+        Implementation is optional.
+        '''
+        edit_html = Freetextresponse.get_resource_string('edit.html')
+        edit_html = edit_html.format(
+            self=self,
+        )
+        fragment = self.build_fragment(
+            html_source=edit_html,
+            paths_css=[
+                'edit.less.min.css',
+            ],
+            paths_js=[
+                'edit.js.min.js',
+            ],
+            fragment_js='FreetextresponseEdit',
+        )
+        return fragment
+
+    @XBlock.json_handler
+    def studio_view_save(self, data, suffix=''):
+        #  pylint: disable=unused-argument
+        '''
+        Save XBlock fields
+        Returns: the new field values
+        '''
+        return self.save_studio_data(data)
+
+    def save_studio_data(self, data):
+        '''
+        Helper method to save the parameters set by the instructor
+        for the given instance of the FreeTextResponse XBlock
+        '''
+        self.display_name = data['display_name']
+        self.prompt = data['prompt']
+        if data['weight'] and int(float(data['weight'])) >= 0:
+            self.weight = int(float(data['weight']))
+        if data['max_attempts'] and int(float(data['max_attempts'])) >= 0:
+            self.max_attempts = int(float(data['max_attempts']))
+        self.display_correctness = data['display_correctness']
+        if data['min_word_count'] and data['max_word_count']:
+            if (
+                    int(float(data['min_word_count'])) >= 0 and
+                    (
+                        int(float(data['min_word_count'])) <=
+                        int(float(data['max_word_count']))
+                    )
+            ):
+                self.min_word_count = int(float(data['min_word_count']))
+            if (
+                    int(float(data['max_word_count'])) >= 0 and
+                    (
+                        int(float(data['max_word_count'])) >=
+                        int(float(data['min_word_count']))
+                    )
+            ):
+                self.max_word_count = int(float(data['max_word_count']))
+        self.fullcredit_keyphrases = data['fullcredit_keyphrases']
```

Comment:
See note above about storing this as a list. We could clean it up with something like:

```python
if 'full_credit_key_phrases' in data:
    self.full_credit_key_phrases = [
        phrase.strip()
        for phrase in data['full_credit_key_phrases'].split(',')
        if phrase.strip()
    ]
```